src/meta.rst

```diff
@@ -71,39 +71,128 @@ A full description of selectors is
 `in the conda docs <http://conda.pydata.org/docs/building/meta-yaml.html#preprocessing-selectors>`_.
 
 
-Using the Compilation ``toolchain``
------------------------------------
-Packages that need to compile code should add the ``toolchain`` package as a build dependency. This helps set up environment variables to instruct the compilers to do the right thing (e.g. building against the right version of the MacOS SDK).
+Pinning packages
+----------------
+
+Conda-smithy 3.0.0 switches to ``conda-build=3``. In conda-smithy 3.0.0, we use a central configuration file from conda-forge-pinning for the build matrices and versions of specific packages.
+
+When a rerendering happens, conda-smithy will render the recipe using conda-build and output configuration files for each job and save them in a yaml file in ``.ci_support`` folder. For example there's a output configuration file for each OS, each python version, etc.
+
+These output configuration files are stripped to options that are used in the build and therefore a change in the config files in ``.ci_support`` folder implies that there needs to be a new build.
+
+Pinning of packages are handled by the same configuration file and conda-build. This means that packages need not be pinned manually. For eg:
```

Comment:
TODO: Explain that if the package is not in `conda-forge-pinning`, it should be added there.